sematic/ui/packages/common/src/pages/RunDetails/NotesPane.tsx

```diff
@@ -35,13 +17,50 @@ const Notes = styled.section`
     overflow-x: hidden;
     scrollbar-gutter: stable;
     margin-right: -${theme.spacing(2.4)};
+    margin-left: -${theme.spacing(2.4)};
 `;
 
+function getUserName(user: User | null): string {
+    return (!!user && user.first_name) || "Anonymous";
+}
+
 const NotesPane = () => {
+    const { rootRun, isGraphLoading } = useRootRunContext() as ExtractContextType<typeof RootRunContext> & {
+        rootRun: Run;
+    };
+    const { selectedRun } = useRunDetailsSelectionContext();
+    const { user } = useContext(UserContext);
+
+    const { notes, reload: reloadNotes } = usePipelineNotes(rootRun.function_path);
+    const [, submitNoteToServer] = useNoteSubmission();
+
+    const isSubmitting = useRef(false);
+
+    const onSubmit = useCallback(async (content: string) => {
+        if (isSubmitting.current || isGraphLoading) {
+            return;
+        }
+        isSubmitting.current = true;
+
+        await submitNoteToServer({
+            note: {
+                author_id: user?.email || "anonymous@acme.com",
+                note: content,
+                root_id: rootRun.id,
+                run_id: selectedRun!.id,
+            }
+        });
+
+        isSubmitting.current = false;
+        reloadNotes();
+    }, [submitNoteToServer, user, isGraphLoading, selectedRun, reloadNotes, rootRun.id]);
+
     return <>
-        <SubmitNoteSection />
+        <SubmitNoteSection onSubmit={onSubmit} />
         <Notes>
-            {MockNoteData.map((note, index) => <Note key={index} {...note} />)}
+            {(notes || []).map(({ note, run_id, created_at, user }, index) =>
+                <Note key={index} content={note} name={getUserName(user)} runId={run_id}
+                    createdAt={created_at as unknown as string} />)}
         </Notes>
```

Comment:
Positioning the submit form at the top of the Notes panel is very awkward to use. I don't know of any messaging system that does this. Please move it to the bottom.

Reply:
It does feel awkward, especially when the notes are in chronological order. Either we move the submission down or reverse the chronological order, at least one should be done.

What's your opinion on this please @neutralino1 ?

Reply:
Yes the order should be reversed, most recent at the top.

The idea is to make it like a log of messages. Think about comment sections on blog posts. The input field is at the top and most recent comment at the top.

Reply:
Done. Adopted reverse chronological order.